### DISCUSSION TITLE
为什么我们需要NextRankPredictor？

### DISCUSSION THREAD

**Discussion by Koyonomi** [Equim-chan/Mortal#52] [Q&A]
NextRankPredictor被设计从当前步的phi去预测next_kyoku的玩家排名。通过train.py的训练，我们试图最小化预测值与真实玩家排名的误差。但是NextRankPredictor被训练完成后，它的预测值有什么作用？(因为预测值在offline模式下似乎没有对训练结果任何帮助)。抱歉，我在Mortal当前公开的代码中没有发现这一点。

**Comment by Equim-chan** [comment]
它是一个 auxiliary task。

**Reply by Koyonomi** [reply]
它的作用是提升模型泛化性？此外，我注意到NextRankPredictor的任务目标与NAGA牌谱分析的"期待顺位"功能比较类似。

**Reply by Equim-chan** [reply]
对泛化性可能有帮助。NAGA 的“期待顺位”我不太清楚，不过 Mortal 目前的 NextRankPredictor 只预测到下一局开局时的顺位，并且只用来训练而没用在推理。

**Reply by Koyonomi** [reply]
是的，NAGA"期待顺位"功能个人认为“从意义上”近似于NextRankPredictor + GlobalRewardPrediction，即：进行牌谱分析时通过encode后的obs预测next_kyoku开始时的ranks和scores(尽管NextRankPredictor实际没有预测scores)，然后(GRP)利用预测得到的scores(ranks)去预测终局顺位。但GRP仅通过kyoku开始时四名玩家的分数、本场、场供等信息去预测遥远的终局，accuracy是比较低的，更多是用于计算kyoku_rewards。所以我上面提到NextRankPredictor与NAGA期待顺位功能可能具有相似性
